Automattic/wp-calypso
# Plans 2023: Split and export the individual grids (FeaturesGrid, ComparisonGrid) - take 2

Related to https://github.com/Automattic/wp-calypso/issues/78266
Takes over https://github.com/Automattic/wp-calypso/pull/80915

## Proposed Changes

Converts the main package component `plans-grid` into an index of individual parts, exported and assembled at the consuming end. We effectively now export `FeaturesGrid` and `ComparisonGrid` separately, and a single `gridPlans` prop in place of `gridPlansForFeauresGrid`, etc.

### Additional Notes

- There is some repetition in the "wrappers" exported from the main index file, and that's intentional. All of that will gradually shrink to simply reflected imports as we clean up more dependencies i.e. there will be no need for these wrappers altogether.
- The `ComparisonGrid` carries its title along, which is something incredibly small that we can handle in a follow-up. 
- A props refactor should also happen on top of this PR. A separate issue exists: https://github.com/Automattic/wp-calypso/issues/82335
- Some performance concerns are being handled in a separate PR. For the most part, things should be functioning reasonably.

## Testing Instructions

- Confirm `/start` and `/plans/[ paid | free ]` render as before
- Confirm tailored flows (newsletter, wooexpress) work as before
- Confirm plan-term toggle works correctly



## Pre-merge Checklist
